src/dotty/tools/dotc/transform/ClassTags.scala

```diff
@@ -0,0 +1,65 @@
+package dotty.tools.dotc
+package transform
+
+import core._
+import TreeTransforms._
+import Contexts.Context
+import Flags._
+import SymUtils._
+import Symbols._
+import SymDenotations._
+import Types._
+import Decorators._
+import DenotTransformers._
+import StdNames._
+import NameOps._
+import ast.Trees._
+import dotty.tools.dotc.ast.tpd
+import dotty.tools.dotc.core.Constants.Constant
+import util.Positions._
+import Names._
+import collection.mutable
+
+/** This phase replaces calls to DottyPredef.classTag by code that synthesizes appropriate ClassTag
+  */
+class ClassTags extends MiniPhaseTransform with IdentityDenotTransformer { thisTransform =>
+  import ast.tpd._
+
+  private var classTagCache: Symbol = null
+  private var typeTagCache: Symbol  = null
```

Comment:
I usually use myClassTag/myTypeTag for private variables, includes variables serving as a cache.